modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/GridDhtTxFinishFuture.java

```diff
@@ -239,7 +241,21 @@ public void onResult(UUID nodeId, GridDhtTxFinishResponse res) {
                     try {
                         boolean nodeStopping = X.hasCause(err, NodeStoppingException.class);
 
-                        this.tx.tmFinish(err == null, nodeStopping || cctx.kernalContext().failure().nodeStopping(), false);
+                        if (cctx.consistentCutMgr() != null && this.tx.currentPrepareFuture() != null) {
```

Comment:
Can we

a. rework logic to avoid `cctx.consistentCutMgr() != null` checks.
b. use NoOp or disabled implementation of cut manager in case it can't be used.

Reply:
Or, at least, extract wrap logic to the utility method.